scripts/zf_make_official_srpm

```diff
@@ -10,88 +10,89 @@ fail() { log "$*"; exit 1; }
 try()  { "$@" || fail "FAILED: $*"; }
 
 
-usage_msg() {
-  err
-  err "usage:"
-  err "  $me <tcpdirect-tarball>"
-  err
-  err "options:"
-  err
-}
-
+my_dir=$(cd "$(dirname "$0")" && /bin/pwd)
+top_dir=$(dirname "${my_dir}")
+
+copy_to_tmpdir="
+versions.yaml
+Makefile
+Makefile.onload
+Makefile-top.inc
+doc
+src
+scripts
+"
 
 usage() {
-  usage_msg
+  echo
+  echo "usage:"
+  echo "  $(basename "$0") [options]"
+  echo
+  echo "options:"
+  echo "  --version <version>"
+  echo
   exit 1
 }
 
 
-logtry() {
-  log "$*"
-  try "$@"
+cleanup() {
+  echo "Deleting ${tmpdir}"
+  rm -rf "${tmpdir}"
 }
 
 
-cleanup() {
-  [ -d "$tmpd" ] && rm -rf "$tmpd"
+copy_files_to_tmpdir() {
+  try mkdir -p "${zf_tmpdir}"
+  # shellcheck disable=SC2086
+  try tar -cz -C "${top_dir}" ${copy_to_tmpdir} | tar xz -C "${zf_tmpdir}"
+}
+
+
+patch_tmpdir_files() {
+  try sed -i \
+    -e "s/pkgversion-DEFINITION/%define pkgversion ${version}/g" \
+    "${zf_tmpdir}/scripts/tcpdirect_misc/tcpdirect.spec"
+}
+
+
+make_tarball() {
+  tarball="${zf_package}.tgz"
+
+  try tar -czf "${tarball}" -C "${tmpdir}" "${zf_package}"
+  try md5sum "${tarball}" > "${tarball}.md5"
+}
+
+
+make_srpm() {
+  tarball="${zf_package}.tgz"
```

Comment:
Given this is defined in both `make_tarball` and `make_srpm`, it could be a good idea to just lump this with the other global definitions (e.g., right after `zf_package`).

Reply:
Yep, that sounds good.

(FWIW, I'm not keen on using global variables this way, but it makes perfect sense for smaller scripts like this!)